src/openrct2-ui/windows/SavePrompt.cpp

```diff
@@ -143,8 +215,8 @@ rct_window* WindowSavePromptOpen()
         log_warning("Invalid save prompt mode %u", prompt_mode);
         return nullptr;
     }
-    window = WindowCreateCentred(
-        width, height, &window_save_prompt_events, WindowClass::SavePrompt, WF_TRANSPARENT | WF_STICK_TO_FRONT);
+    window = WindowCreate<SavePromptWindow>(
+        WindowClass::SavePrompt, width, height, WF_TRANSPARENT | WF_STICK_TO_FRONT | WF_CENTRE_SCREEN);
 
     window->widgets = widgets;
```

Comment:
There is a loooot of logic here on this `WindowSavePromptOpen` that should be moved to the new window `OnOpen` function instead.